implementation/fault-tolerance/src/main/java/io/smallrye/faulttolerance/CircuitBreakerMaintenanceRegistry.java

```diff
@@ -0,0 +1,67 @@
+package io.smallrye.faulttolerance;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.spi.InjectionPoint;
+import javax.inject.Singleton;
+
+import io.smallrye.faulttolerance.api.CircuitBreakerMaintenance;
+import io.smallrye.faulttolerance.api.CircuitBreakerName;
+import io.smallrye.faulttolerance.api.CircuitBreakerState;
+import io.smallrye.faulttolerance.core.circuit.breaker.CircuitBreaker;
+
+@Singleton
+public class CircuitBreakerMaintenanceRegistry {
+    private final ConcurrentMap<String, CircuitBreaker<?>> registry = new ConcurrentHashMap<>();
+
+    void register(String circuitBreakerName, CircuitBreaker<?> circuitBreaker) {
+        registry.putIfAbsent(circuitBreakerName, circuitBreaker);
+    }
+
+    @Produces
+    @CircuitBreakerName("")
+    @Dependent
+    CircuitBreakerMaintenance produce(InjectionPoint injectionPoint) {
+        // TODO use injectionPoint.getQualifiers instead?
+        String name = injectionPoint.getAnnotated().getAnnotation(CircuitBreakerName.class).value();
```

Comment:
Hi @manovotn, could you please advise what's best here? Would both `injectionPoint.getAnnotated().getAnnotation(...)` and `injectionPoint.getQualifiers()` work? Is there a Quarkus concern? Thanks!

Reply:
Hi @Ladicek 

From purely spec perspective, both are correct and Quarkus will allow you to do this as well. However, if you are paranoid then switch to `injectionPoint.getQualifiers()`.
The reason is that Quarkus doesn't have a full notion of `Annotated` and [creates it just ad-hoc](https://github.com/quarkusio/quarkus/blob/master/independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/CurrentInjectionPointProvider.java#L77-L84) for this very use case (because we want to support `InjectionPoint` but we don't have the whole hierarchy of `Annotated` in ArC). Though like I said - this should work in Quarkus as well ;-)

Reply:
Cool, thanks! I'll switch to `getQualifiers` then.